cli/command/task.go

```diff
@@ -10,6 +17,219 @@ func NewTaskCommand(b *CommandBase) *TaskCommand {
+					},
+					cli.StringSliceFlag{
+						Name:  "env",
+						Usage: "environment variable override in format 'CONTAINER:VAR=VAL' (can be specified multiple times)",
+					},
+				},
+			},
+			{
+				Name:      "delete",
+				Usage:     "delete a Task",
+				ArgsUsage: "TASK_NAME",
+				Action:    t.delete,
+			},
+			{
+				Name:      "list",
+				Usage:     "list all Tasks",
+				Action:    t.list,
+				ArgsUsage: " ",
+			},
+			{
+				Name:      "read",
+				Usage:     "describe a Task",
+				Action:    t.read,
+				ArgsUsage: "TASK_NAME",
+			},
+			{
+				Name:      "logs",
+				Usage:     "get the logs for a task",
+				Action:    t.logs,
+				ArgsUsage: "TASK_NAME",
+				Flags: []cli.Flag{
+					cli.IntFlag{
+						Name:  "tail",
+						Usage: "number of lines from the end to return",
+					},
+					cli.StringFlag{
+						Name:  "start",
+						Usage: "the start of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+					cli.StringFlag{
+						Name:  "end",
+						Usage: "the end of the time range to fetch logs (format: YYYY-MM-DD HH:MM)",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (t *TaskCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "ENVIRONMENT", "TASK_NAME", "DEPLOY")
+	if err != nil {
+		return err
+	}
+
+	overrides, err := parseOverrides(c.StringSlice("env"))
+	if err != nil {
+		return err
+	}
+
+	environmentID, err := t.resolveSingleEntityIDHelper("environment", args["ENVIRONMENT"])
+	if err != nil {
+		return err
+	}
+
+	deployID, err := t.resolveSingleEntityIDHelper("deploy", args["DEPLOY"])
+	if err != nil {
+		return err
+	}
+
+	req := models.CreateTaskRequest{
+		TaskName:           args["TASK_NAME"],
+		EnvironmentID:      environmentID,
+		DeployID:           deployID,
+		ContainerOverrides: overrides,
+	}
+
+	jobID, err := t.client.CreateTask(req)
+	if err != nil {
+		return err
+	}
+
+	return t.waitOnJobHelper(c, jobID, "creating", func(taskID string) error {
+		task, err := t.client.ReadTask(taskID)
+		if err != nil {
+			return err
+		}
+
+		return t.printer.PrintTasks(task)
+	})
+}
+
+func (t *TaskCommand) delete(c *cli.Context) error {
+	return t.deleteHelper(c, "task", func(taskID string) (string, error) {
+		return t.client.DeleteTask(taskID)
+	})
+}
+
+func (t *TaskCommand) list(c *cli.Context) error {
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	return t.printer.PrintTaskSummaries(taskSummaries...)
+}
+
+func (t *TaskCommand) read(c *cli.Context) error {
+	taskSummaries, err := t.client.ListTasks()
+	if err != nil {
+		return err
+	}
+
+	taskExists := map[string]bool{}
+	for _, taskSummary := range taskSummaries {
+		taskExists[taskSummary.TaskID] = true
+	}
+
+	args, err := extractArgs(c.Args(), "TASK_NAME")
+	if err != nil {
+		return err
+	}
+
+	taskIDs, err := t.resolver.Resolve("task", args["TASK_NAME"])
+	if err != nil {
+		return err
+	}
+
+	tasks := make([]*models.Task, len(taskIDs))
```

Comment:
Can't use `make` in this way here since we aren't sure every task will be read. 

However, you can use `make([]*models.Task, 0, len(taskIDs))`. This means "create a new slice of length 0, but prepare to allocate up to `len(taskIDs)` worth of memory". With that method, you will need to use `append` to the slice instead of `tasks[i] = ...`